crates/interpreter/src/exec.rs

```diff
@@ -1041,10 +1045,18 @@ impl<'m> Thread<'m> {
         if i == 0 {
             return self.exit_frame();
         }
-        let values = core::mem::take(self.values());
         let frame = self.frame();
-        let Label { arity, kind, .. } = frame.labels.drain(i ..).next().unwrap();
-        self.values().extend_from_slice(&values[values.len() - arity ..]);
+        // TODO: Store prefix sums of the labels' values_cnt in Frame to quickly compute
+        // label_values_cnt_before.
+        let label_values_cnt_before: usize =
+            frame.labels[0 .. i].iter().map(|label| label.values_cnt).sum();
+        let Label { arity, kind, values_cnt } = frame.labels.drain(i ..).next().unwrap();
+        // TODO: Is the following line necessary?
+        self.values().drain(label_values_cnt_before + values_cnt ..);
+        self.values()
+            .drain(label_values_cnt_before .. label_values_cnt_before + values_cnt - arity);
```

Comment:
@ia0 Would like to get some clarification that will help me fix the remaining 9 failing tests. Thanks!

1) From `values[values.len() - arity ..]` in line 1047 in the original code, it looks like `arity` is stored after the other values. Two questions:
1.1) I don't see how `arity` is stored in `values` in the first place, by reading [push_label](https://github.com/google/wasefire/blob/main/crates/interpreter/src/exec.rs#L1029) and [Frame::new](https://github.com/google/wasefire/blob/main/crates/interpreter/src/exec.rs#L1408).
1.2) If `arity` is stored in `values`, my "intuition" is that it should be stored in front because it could represent the result.

2) See the TODO on line 1054 in the new code. Do we also need to remove the labels after the `i`-th one? Why or why not?